src/main/scala/decider/Decider.scala

```diff
@@ -87,40 +87,49 @@ trait DefaultDeciderProvider extends VerifierComponent { this: Verifier =>
   def identifierFactory: IdentifierFactory
 
   object decider extends Decider with StatefulComponent {
-    private var z3: Z3ProverStdIO = _
+    private var proverStdIO: ProverStdIO = _
     private var pathConditions: PathConditionStack = _
 
 //    private var _freshFunctions: InsertionOrderedSet[FunctionDecl] = _ /* [BRANCH-PARALLELISATION] */
 //    private var _freshMacros: Vector[MacroDecl] = _
 
-    def prover: Prover = z3
+    def prover: ProverStdIO = proverStdIO
 
     def pcs: PathConditionStack = pathConditions
 
 //    def setPcs(other: PathConditionStack) = { /* [BRANCH-PARALLELISATION] */
 //      pathConditions = other
 //      pathConditions.assumptions foreach prover.assume
 //    }
+    private def getProverStdIO(prover: String): ProverStdIO = prover match {
```

Comment:
Could this be an enum at this point (created when parsing config flags) rather than a `String`?

Reply:
Letting the factory handle strings instead of anything else was primarily motivated by its simplicity. 

We could create an enum and an extra ValueConverter in Config.scala that converts the strings to an enum but wouldn't we still need a factory that just matches on the enum to create new ProverStdIO objects? To me it seems like this would add another place in the code to update when introducing additional SMT solvers with no added benefits. Admittedly, I am very new to Scala so there are likely advantages that I do not see or a way to easily create a factory with enums.

If you have any suggestions on a clean/nice way to do this, I would be more than happy to introduce the changes if I have time.

Reply:
Enums (or sealed traits) have the advantage over strings that the compiler can perform exhaustiveness checks for the former. It is therefore good practice to turn strings into enums as early as possible (i.e. in the command-line parser). Another disadvantage of strings is that you always need to account for upper-/lowercase spelling.

Reply:
Great, thanks for the explanation.
I will introduce enums instead of the string factory if I have time. :)